clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -609,7 +607,18 @@ private void handleProduceResponse(ClientResponse response, Map<TopicPartition,
                                 .collect(Collectors.toList()),
                             p.errorMessage(),
                             p.currentLeader());
-                    ProducerBatch batch = batches.get(tp);
+                    // Version 13 drop topic name and add support to topic id.
+                    // We need to find batch based on topic id and partition index only as
+                    // topic name in the response might be empty.
```

Comment:
How about the following?

                    // Version 13 drop topic name and add support to topic id.
                    // We need to find batch based on topic id and partition index only as
                    // topic name in the response will be empty.
                    // For older versions, topic id is zero and we will find the batch based on the topic name.